core/src/test/scala/com/rallyhealth/vapors/factfilter/evaluator/LogicalExprSpec.scala

```diff
@@ -0,0 +1,334 @@
+      val q = {
+        and(or(T, T), or(T, T))
+      }
+      shouldBeTrue(evalOutput(q))
+    }
+
+    "return 'true' for two nested 'not false or's in an 'and'" in {
+      val q = {
+        and(not(or(F, F)), not(or(F, F)))
+      }
+      shouldBeTrue(evalOutput(q))
+    }
+
+    "return 'false' for a nested 'false or' before some 'true' expressions" in {
+      val q = {
+        and(or(F, F), T, T)
+      }
+      shouldBeFalse(evalOutput(q))
+    }
+
+    "return 'true' for a nested 'not true or' before some 'true' expressions" in {
+      val q = {
+        and(not(or(F, T)), T, not(F))
+      }
+      shouldBeFalse(evalOutput(q))
+    }
+
+    "return 'false' for a nested 'false or' after some 'true' expressions" in {
+      val q = {
+        and(T, or(T, F), or(F, F))
+      }
+      shouldBeFalse(evalOutput(q))
+    }
+
+    "return 'false' for a nested 'not true or' after some 'true' expressions" in {
+      val q = {
+        and(T, not(F), or(not(T), T), or(F, not(T)))
+      }
+      shouldBeFalse(evalOutput(q))
+    }
+
+    "return 'true' for a nested 'false and' in an 'or'" in {
+      val q = {
+        or(F, and(F, F), and(T, T), and(T, F))
+      }
+      shouldBeTrue(evalOutput(q))
+    }
+
+    "return 'false' for the negation of a nested 'false and' in an 'or'" in {
+      val q = {
+        not(or(F, and(F, F), not(and(T, T)), and(T, not(F), not(T))))
+      }
+      shouldBeTrue(evalOutput(q))
+    }
+
+    "return 'true' for a complex structure" in {
+      val q = {
+        or(F, and(F, T), or(F, F), and(or(T, F), F), or(and(T, F), T))
+      }
+      shouldBeTrue(evalOutput(q))
+    }
+
+    "return 'false' for a complex structure" in {
+      val q = {
+        not(
+          or(
+            F,
+            not(T),
+            and(F, T),
+            or(F, F),
+            not(and(T, T)),
+            not(or(T, F)),
+            not(not(F)),
+            and(or(T, F), F, not(T)),
+            or(F, not(T), and(T, F), not(and(T, not(F)))),
+          ),
+        )
+      }
+      shouldBeTrue(evalOutput(q))
+    }
+  }
+
+  private abstract class DslLogicOpBuilder {
+
+    def andBuilder[R : Conjunction : ExtractBoolean]: LogicOpBuilder[R]
+
+    def orBuilder[R : Disjunction : ExtractBoolean]: LogicOpBuilder[R]
+
+    def notBuilder[R : Negation]: UnaryLogicOpBuilder[R]
+  }
+
+  def validDslLogicalOperators(builder: DslLogicOpBuilder): Unit = {
+
+    "operating on boolean results" should {
+
+      behave like validLogicalOperators[Boolean](
+        builder.andBuilder,
+        builder.orBuilder,
+        builder.notBuilder,
+        trueBuilder = const(true),
+        falseBuilder = const(false),
+        facts = JoeSchmoe.facts,
+        assertTrue = { implicit pos => o =>
+          assert(o.value)
+          // TODO: Is is sufficient to only test all the facts or no facts?
+          assertResult(FactsMatch(JoeSchmoe.facts))(o.evidence)
+        },
```

Comment:
I think so...? None of the tests differentiate between fact types. It might be possibly to test this in the same way as EmbeddedExprSpec by using specific facts and combining them in ways that we expect this to work. I'll probably tackle this in a subsequent PR.